### ISSUE TITLE
No matching distribution found for tensorflow-text

### ISSUE THREAD

**Issue by QUENTINNE** [tensorflow/text#89]
**I failed to install tensorflow-text.** 

When I enter `pip install -U tensorflow-text`

There was an error: 

> Could not find a version that satisfies the requirement tensorflow-text (from versions: )
> No matching distribution found for tensorflow-text

- Python 3.5.4  [MSC v.1900 64 bit (AMD64)] on win32
- Tensorflow 2.0.0rc0

**Comment by broken** [comment]
There are a few of things we are trying to squeeze in, but we will be releasing soon.

**Comment by broken** [comment]
To elaborate, Tensorflow has changed the version of gcc they compiled against starting with 2.0.0rc0. We are updating our builds to match and getting in a couple of other updates (eg. sentencepiece, fix for #84, etc) in. I expect a release soon, but users can use 2.0.0b2 in the meantime.

Apart from this, we do not have a Windows ready release. There have been complications getting the dynamic libraries to build and then deploy with the normal tf pip package. This is an ongoing issue #44 and one we still plan on addressing.

**Comment by broken** [comment]
It was much more difficult to update the build environment to match core TF than I expected. However, we are finally in a maintainable state again and have releases for 1.15 and 2.0.

**Comment by mborsetti** [comment]
@broken nice talk at TensorFlow World, and great work here!

You say above, in a thread about Windows binaries, "have releases for 1.15 and 2.0", yet I don't see them at https://pypi.org/project/tensorflow-text/2.0.0rc0/#files

Would love to use the tool ... on Windows!  Any chance they can be uploaded to PiPy?

**Comment by broken** [comment]
Thanks; and my apologies about this confusion. When I revisited this issue I overlooked you had win32 as the platform. Unfortunately, Windows is still an open issue, as we don't have builds for it yet. We've left #44 open to track it, and I probably should've marked this as a duplicate originally.

We very much want to support Windows, but honestly have had difficulty finding the eng time to work through the build issues. I cannot give a good ETA as much as I wish I could, as there are a large number of TF users on Windows that we want to support.

**Comment by antoinecomp** [comment]
I have the same error message and I'm on windows. I can't use `tensorflow-text` yet if I've well understood the posts above, right?

tensorflow           2.1.0
Python                3.6.7

**Comment by mzeidhassan** [comment]
same thing here. I cannot use it on Windows.

```
ERROR: Could not find a version that satisfies the requirement tensorflow_text (from versions: none)
ERROR: No matching distribution found for tensorflow_text
```

I hope Windows will be supported soon.

**Comment by jeanru** [comment]
I have the same issue, referring to pypi link https://pypi.org/project/tensorflow-text/#files , I think maybe I have to change to other OS platform. Also I hope it works on Windows someday.

**Comment by leo-gan** [comment]
I have the same issue as @mzeidhassan 
Sept-Oct-Dec-Jan-Feb-Mar and we still here.

**Comment by thiziri** [comment]
I have the same issue too:

- Windows 64 + python 3.6 
- Installed successfully: tensorflow_hub, tensorflow2, sentencepiece
- Not supported: TensorFlow-text

So, when iI try loading models of the  universal-sentence-encoder-multilingual, I got the followin error:

> NotFoundError: Op type not registered 'SentencepieceEncodeSparse' in binary running on ... Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.

Isn't it solved yet?

**Comment by soumayan** [comment]
still this problem exists for windows,any updates for windows users??

**Comment by zzj0402** [comment]
Kaggle notebook:
```
!pip install tensorflow_text
```
Same error.

**Comment by mzeidhassan** [comment]
We need a solution please. This is a show-stopper to use Universal Sentence Encoder.

**Comment by zzj0402** [comment]
> Kaggle notebook:
> 
> ```
> !pip install tensorflow_text
> ```
> 
> Same error.

Solved by toggling gpu.

**Comment by mzeidhassan** [comment]
@zzj0402  Can you please explain? I have only one GPU on my Windows 10 machine. How do you exactly toggle gpu? Thanks in advance!

**Comment by evyasonov** [comment]
I have the same problem

**Comment by pathway** [comment]
Same problem: ubuntu 18.04 python 3.8.3 tensorflow-gpu-2.2.0
Installed tf2 via: conda install tensorflow-gpu

**Comment by ManuMahadevaswamy** [comment]
'model.ckpt-1500000.data-00000-of-00001' , is this the hybrid model which is trained on 'c4+HugeNews' datasets?
'c4.unigram.newline.10pct.96000.model',--is this the model trained on only c4 data?

**Comment by ManuMahadevaswamy** [comment]
> 'model.ckpt-1500000.data-00000-of-00001' , is this the hybrid model which is trained on 'c4+HugeNews' datasets?
> 'c4.unigram.newline.10pct.96000.model',--is this the model trained on only c4 data?

Can anyone explain this please?

**Comment by bhanuprakashd** [comment]
same problem i too facing. Can someone help to solve this i have 2.2.0 version tensorflow

**Comment by HumanCodingMachine** [comment]
UGH! I've been trying on Mac, Linux, and Windows, and nothing has worked.

**Comment by xepozz** [comment]
I have the same error on Mac.
Python 3.8
TF 2.6.2
@broken could you please reopen this issue?

**Comment by broken** [comment]
@xepozz I see a Mac distribution for tf-text 2.6.x on pypi (https://pypi.org/project/tensorflow-text/2.6.0/#files), so likely you have a M1 mac which we currently do not have prebuilt packages for. We cannot support these until TF does (see  #654 & #538); however, you should be able to build tf-text yourself and use it.

If this is not correct, can you create a new issue? Your error message may be the same, but it is most likely a different problem than the original.
